server/datastore/mysql/software.go

```diff
@@ -3279,6 +3279,10 @@ func (ds *Datastore) ListHostSoftware(ctx context.Context, host *fleet.Host, opt
 			s.Version = installedTitle.Version
 			s.BundleIdentifier = installedTitle.BundleIdentifier
 		}
+		// Override the status; if there's a pending re-install, we should show that status.
+		if hs, ok := byVPPAdamID[*s.VPPAppAdamID]; ok {
```

Comment:
Any chance that `*s.VPPAppAdamID` could be nil here?

Reply:
Not sure, but better safe than sorry! I'll add a check.